File: src/main/java/org/intellij/sdk/language/AngelScriptReferenceContributor.java
package org.intellij.sdk.language;

import com.intellij.patterns.PlatformPatterns;
import com.intellij.psi.*;
import com.intellij.util.ProcessingContext;
import org.jetbrains.annotations.NotNull;


public class AngelScriptReferenceContributor extends PsiReferenceContributor {
	@Override
	public void registerReferenceProviders( @NotNull PsiReferenceRegistrar registrar ) {
		registrar.registerReferenceProvider(
			PlatformPatterns.psiElement( PsiLiteralExpression.class ),
			new PsiReferenceProvider() {
				@Override
				public PsiReference @NotNull [] getReferencesByElement( @NotNull PsiElement element, @NotNull ProcessingContext context ) {
					PsiLiteralExpression literalExpression = (PsiLiteralExpression) element;
					if ( literalExpression.getValue() instanceof String ) {
						return new PsiReference[] { new AngelScriptReference( element ) };
					}
					return PsiReference.EMPTY_ARRAY;
				}
			}
		);
	}

}

File: src/main/kotlin/com/enderzombi102/css/lang/angelscript/AngelScriptReferenceContributor.kt
package com.enderzombi102.css.lang.angelscript

import com.intellij.patterns.PlatformPatterns
import com.intellij.psi.*
import com.intellij.util.ProcessingContext

class AngelScriptReferenceContributor : PsiReferenceContributor() {
	override fun registerReferenceProviders( registrar: PsiReferenceRegistrar ) {
		registrar.registerReferenceProvider(
			PlatformPatterns.psiElement( PsiLiteralExpression::class.java ),
			object : PsiReferenceProvider() {
				override fun getReferencesByElement( element: PsiElement, context: ProcessingContext ) =
					if ( ( element as PsiLiteralExpression ).value is String )
						arrayOf( AngelScriptReference(element) )
					else
						PsiReference.EMPTY_ARRAY
			}
		)
	}
}